addons/website_crm/controllers/main.py

```diff
@@ -60,6 +60,12 @@ def _handle_website_form(self, model_name, **kwargs):
     def insert_record(self, request, model, values, custom, meta=None):
         is_lead_model = model.model == 'crm.lead'
         if is_lead_model:
+            visitor = request.env['website.visitor']._get_visitor_from_request()
+            visitor_partner = visitor.partner_id
+            if visitor and visitor_partner:
+                fmt_partner_number = visitor_partner._phone_format(visitor_partner.phone)
+                if fmt_partner_number == values['phone'] and visitor_partner.email == values['email_from']:
```

Comment:
I think values['phone'] should also be formatted to have a comparison that makes sense. And for that you need a country for the visitor.

Reply:
@tde-banana-odoo  Yes of course, but of what I saw (maybe I got confused) this is done in _handle_website_form, that in turns calls this method so values['phone'] is already formatted? :)

Reply:
Ah ok. Formatted using international format, or E164 ? Just be sure formatting is done on same basis :) .

Probably worth a comment by the way to tell that "phone" is already formatted.

Reply:
Good point, will do. When going through phone_validation (or other modules), I noticed in some parts of the code that sometimes E164 was used. But here, I made sure that both are done in same format : international :)

Reply:
We use E164 format to store in phone_sanitized as it is more a standard-based formatting. International is more user friendly I think. If both are international that's ok indeed :) .